sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
         return join;
     }
 
+
+    /**
+     * Build outputs of joins not in the outer level of the tree
+     * based on the where queries that must be applied later on
+     * in case the current outputs are empty (case of count(*))
+     * e.g.:
+     * select count(*) from t1, t2, t3 where t1.a = t2.b and t2.b = t3.c
+     *
+     *                   join
+     * outputs=t1[id]   /   \
+     *      |          /     \
+     *      +------> join    t3
+     *               / \
+     *              /   \
+     *             t1   t2
+     */
+    private static void buildJoinOutputs(Map<Set<QualifiedName>, Symbol> splitQuery,
+                                         Iterator<QualifiedName> it,
+                                         final QualifiedName leftName,
+                                         final QualifiedName rightName,
+                                         QuerySpec newQuerySpec) {
+        if (it.hasNext() && newQuerySpec.outputs().isEmpty()) {
+            List<Symbol> fields = new ArrayList<>();
+            for (Map.Entry<Set<QualifiedName>, Symbol> entry : splitQuery.entrySet()) {
+                if (entry.getKey().contains(leftName) || entry.getKey().contains(rightName)) {
```

Comment:
might want to assign `entry.getKey()` to a local variable (used twice).  Could do the same for entry.getValue() just for the sake of giving it a name (`joinCondition` probably?)